### PR TITLE
Modify sdk for kernel module

### PR DESCRIPTION
**Pull request by Yoshihiro-Okada-ML**
# Purpose of pull request

The SDK provided with EMLinux3.x can't build out-of-tree kernel module due to lack of kernel source code and some environment settings.  
This commit fix this issue.

# Test
## How to test

1. Prepare sample driver code. For example
```
$ cat Makefile 
KERNELDIR     ?= $(SDKTARGETSYSROOT)/usr/src/kernel
KERNELRELEASE ?= $(shell cat $(KERNELDIR)/include/config/kernel.release)

include $(KERNELDIR)/.config

TARGET = hello-procfs.ko

obj-m += hello-procfs.o

all:
	make -C $(KERNELDIR) M=$(PWD)

clean:
	make -C $(KERNELDIR) M=$(PWD) clean

install:
	mkdir -p $(DESTDIR)/lib/modules/$(KERNELRELEASE)/extra
	install -t $(DESTDIR)/lib/modules/$(KERNELRELEASE)/extra $(TARGET)
$
$ cat hello-procfs.c 
#include <linux/module.h>
#include <linux/kernel.h>
#include <linux/proc_fs.h>
#include <linux/init.h>
#include <asm/uaccess.h>
#include <linux/uaccess.h>

#define PROCFS_NAME "hello-procfs"

struct proc_dir_entry;

static char *message = "Hello, /proc module!\n";
static struct proc_dir_entry *entry;

static ssize_t procfs_read(struct file *fp, char __user *user, size_t size, loff_t *offset)
{
  int len, ret;

  pr_info(KERN_INFO "file operation .read (/proc/%s) called\n", PROCFS_NAME);

  len = strlen(message);
  if (len > size) {
    pr_info(KERN_INFO "message(%s) is truncated\n", message);
  }
  ret = copy_to_user(user, message, size<len ? size:len);

  return len;
}

struct proc_ops pops = {
  .proc_read = procfs_read
};

static int __init hello_procfs_init(void)
{
  entry = proc_create(PROCFS_NAME, 0644, NULL, &pops);

  if (entry == NULL) {
    remove_proc_entry(PROCFS_NAME, NULL);
    pr_info(KERN_ALERT "Error: Could not initialize /proc/%s\n",
            PROCFS_NAME);
    return -ENOMEM;
  }

//  proc_set_user(entry, 0, 0);

  pr_info(KERN_INFO "/proc/%s created\n", PROCFS_NAME);
  return 0;
}

static void __exit hello_procfs_exit(void)
{
  proc_remove(entry);
  pr_info(KERN_INFO "/proc/%s removed\n", PROCFS_NAME);
}

module_init(hello_procfs_init);
module_exit(hello_procfs_exit);

MODULE_LICENSE("MIT");
MODULE_AUTHOR("MIRACLE LINUX CORPORATION");
MODULE_DESCRIPTION("Sample /proc/hello-procfs Module");
```
2. Start SDK environment.
 $ . [SDK dir]/emlinux-image-base-sdk-emlinux-bookworm-ls1046ardb/environment-setup-ls1046ardb-emlinux-bookworm 
3. Enter sample driver code directory and type following command.
 $ make 

Before applying this commit, you can see following error.
```
$ make
Makefile:4: /usr/src/kernel/.config: No such file or directory
make: *** No rule to make target '/usr/src/kernel/.config'.  Stop.
```

After applying this commit, you can get compiled kernel module file.
```
$ make
make -C /home/okada/sdk-test/qoriq-sdk/emlinux-image-base-sdk-emlinux-bookworm-ls1046ardb/usr/src/kernel M=/home/okada/tmp/sdk-sample/hello-proc-module_6.1
make[1]: Entering directory '/home/okada/sdk-test/qoriq-sdk/emlinux-image-base-sdk-emlinux-bookworm-ls1046ardb/usr/src/linux-headers-6.1.62-cip9'
warning: the compiler differs from the one used to build the kernel
  The kernel was built by: aarch64-linux-gnu-gcc (Debian 12.2.0-14) 12.2.0
  You are using:           aarch64-linux-gnu-gcc-12.bin (Debian 12.2.0-14) 12.2.0
  CC [M]  /home/okada/tmp/sdk-sample/hello-proc-module_6.1/hello-procfs.o
  MODPOST /home/okada/tmp/sdk-sample/hello-proc-module_6.1/Module.symvers
  CC [M]  /home/okada/tmp/sdk-sample/hello-proc-module_6.1/hello-procfs.mod.o
  LD [M]  /home/okada/tmp/sdk-sample/hello-proc-module_6.1/hello-procfs.ko
make[1]: Leaving directory '/home/okada/sdk-test/qoriq-sdk/emlinux-image-base-sdk-emlinux-bookworm-ls1046ardb/usr/src/linux-headers-6.1.62-cip9'
```

### CODE REVIEW

File: recipes-devtools/sdk-files/files/environment-setup-template.tmpl [hunk 1]
```diff
@@ -21,3 +21,9 @@ export CFLAGS="${TARGET_CFLAGS}"
 export CXXFLAGS="${TARGET_CXXFLAGS}"
 export CPPFLAGS="${TARGET_CXXFLAGS}"
 export LDFLAGS="${TARGET_LDFLAGS}"
+
+export ARCH="${DISTRO_ARCH}"
```
Comment (masami256) [inline recipes-devtools/sdk-files/files/environment-setup-template.tmpl]: I use qemu-amd64 as MACHINE. Then, I got following error.

```
build@d5b543c5d372:~/work/sdk-sample/hello-proc-module_6.1$ make
make -C /home/build/work/sdktest/emlinux-image-base-sdk-emlinux-bookworm-qemu-amd64/usr/src/kernel M=/home/build/work/sdk-sample/hello-proc-module_6.1
make[1]: Entering directory '/home/build/work/sdktest/emlinux-image-base-sdk-emlinux-bookworm-qemu-amd64/usr/src/linux-headers-6.1.67-cip12'
Makefile:775: arch/amd64/Makefile: No such file or directory
make[1]: *** No rule to make target 'arch/amd64/Makefile'.  Stop.
make[1]: Leaving directory '/home/build/work/sdktest/emlinux-image-base-sdk-emlinux-bookworm-qemu-amd64/usr/src/linux-headers-6.1.67-cip12'
make: *** [Makefile:11: all] Error 2
```

according to the error log,  "'arch/amd64/Makefile" is wrong architecture name. it should be x86.

```
build@d5b543c5d372:~/work/sdk-sample/hello-proc-module_6.1$  ARCH=x86  make
make -C /home/build/work/sdktest/emlinux-image-base-sdk-emlinux-bookworm-qemu-amd64/usr/src/kernel M=/home/build/work/sdk-sample/hello-proc-module_6.1
make[1]: Entering directory '/home/build/work/sdktest/emlinux-image-base-sdk-emlinux-bookworm-qemu-amd64/usr/src/linux-headers-6.1.67-cip12'
warning: the compiler differs from the one used to build the kernel
  The kernel was built by: gcc (Debian 12.2.0-14) 12.2.0
  You are using:           x86_64-linux-gnu-gcc-12.bin (Debian 12.2.0-14) 12.2.0
  CC [M]  /home/build/work/sdk-sample/hello-proc-module_6.1/hello-procfs.o
  MODPOST /home/build/work/sdk-sample/hello-proc-module_6.1/Module.symvers
  CC [M]  /home/build/work/sdk-sample/hello-proc-module_6.1/hello-procfs.mod.o
  LD [M]  /home/build/work/sdk-sample/hello-proc-module_6.1/hello-procfs.ko
make[1]: Leaving directory '/home/build/work/sdktest/emlinux-image-base-sdk-emlinux-bookworm-qemu-amd64/usr/src/linux-headers-6.1.67-cip12'
```

So , ARCH variable should not use DISTRO_ARCH  directory.

Comment (Yoshihiro-Okada-ML) [reply recipes-devtools/sdk-files/files/environment-setup-template.tmpl]: To fix this ARCH issue, I add kernel-arch.bbclass based on linux-custom.inc and modify sdk environment setup file to use this class file.

File: classes/sdk-installer.bbclass [hunk 2]
```diff
@@ -95,9 +95,12 @@ python do_create_sdk_installer() {
 
 do_image_tar[postfuncs] += "do_create_sdk_installer"
 
-ROOTFS_POSTPROCESS_COMMAND:append:class-sdk = " rename_installer_script"
+ROOTFS_POSTPROCESS_COMMAND:append:class-sdk = " rename_installer_script setup_kernel_source_dir setup_kernel_source_dir"
```
Comment (masami256) [inline classes/sdk-installer.bbclass]: @Yoshihiro-Okada-ML Do we need "setup_kernel_source_dir" twice?

Comment (Yoshihiro-Okada-ML) [reply classes/sdk-installer.bbclass]: No, it's simply typo.
Fixed.

**Comment by masami256** [comment]
@Yoshihiro-Okada-ML thanks.

**Pull request merged**